build-settings-logic/src/main/kotlin/atomicfu-dependency-resolution-management.settings.gradle.kts

```
@@ -0,0 +1,87 @@
+import org.gradle.kotlin.dsl.maven
+
+pluginManagement {
+    repositories {
+        mavenCentral()
+        gradlePluginPortal()
+
+        val additionalRepositoryProperty = providers.gradleProperty("kotlin_repo_url")
+        if (additionalRepositoryProperty.isPresent) {
+            maven(url = uri(additionalRepositoryProperty.get()))
+            logger.info("A custom Kotlin repository ${additionalRepositoryProperty.get()} was added")
+        }
+
+        /*
+         * This property group is used to build kotlinx.atomicfu against Kotlin compiler snapshots.
+         * When build_snapshot_train is set to true, mavenLocal and Sonatype snapshots are added to a repository list
+         * (the former is required for AFU and public, the latter is required for compiler snapshots).
+         * DO NOT change the name of these properties without adapting the kotlinx.train build chain.
+         */
+        val buildSnapshotTrainGradleProperty = providers.gradleProperty("build_snapshot_train")
+        if (buildSnapshotTrainGradleProperty.isPresent) {
+            maven(url = uri("https://oss.sonatype.org/content/repositories/snapshots"))
+        }
+    }
+}
+
+dependencyResolutionManagement {
+
+    @Suppress("UnstableApiUsage")
+    repositoriesMode.set(RepositoriesMode.PREFER_SETTINGS)
+
+    @Suppress("UnstableApiUsage")
+    repositories {
+        /*
+        * This property group is used to build kotlinx.atomicfu against Kotlin compiler snapshots.
+        * When build_snapshot_train is set to true, mavenLocal and Sonatype snapshots are added to repository list
+        * (the former is required for AFU and public, the latter is required for compiler snapshots).
+        * DO NOT change the name of these properties without adapting kotlinx.train build chain.
+        */
+        val buildSnapshotTrainGradleProperty = providers.gradleProperty("build_snapshot_train")
+        if (buildSnapshotTrainGradleProperty.isPresent) {
+            maven(url = uri("https://oss.sonatype.org/content/repositories/snapshots"))
+        }
+
+        val additionalRepositoryProperty = providers.gradleProperty("kotlin_repo_url")
+        if (additionalRepositoryProperty.isPresent) {
+            maven(url = uri(additionalRepositoryProperty.get()))
```

The kotlin-dsl has a shorter method of adding repositories, which just accepts a string.

I also think it's a nice idea to add a name to the repositories, so that error messages and Build Scan reports are more human readable (otherwise Gradle generates a non-specific name).

```suggestion
            maven(additionalRepositoryProperty.get()) {
                name = "KotlinDevRepo"
            }
```

---

I see, didn't think about Build Scan reports. It makes sense, thanks!